DLBlob/index.js

```diff
@@ -10,79 +10,45 @@
  * @end
  * ----------------------------------------------------------------------------
  */
- 
-const async = require('async');
-const azure = require('azure');
 
+const AlAzureDlBlob = require('al-azure-collector-js').AlAzureDlBlob;
 const ehubCollector = require('../common/ehub_collector');
 const ehubGeneralFormat = require('../EHubGeneral/format').logRecord;
 
-const CONCURRENT_BLOB_PROCESS_NUM = 20;
-
 function getCollectorFunName(blobName) {
     return blobName.split('/')[1];
 }
 
-var collectorProcessError = function(context, err, messages) {
+const processErrorFun = function(context, err, messages) {
     context.log.error('Error processing batch:', err);
     var skipped = messages.records ? messages.records.length : messages.length;
     return skipped;
 };
 
-function processDLBlob(blobService, context, blob, callback) {
-    context.log.verbose('Processing blob: ', blob.name);
-    var collectorFormatFun;
-    
+function processBlob(context, blob, dlblobText, callback) {
+    var formatFun;
     switch(getCollectorFunName(blob.name)) {
         case 'ehubgeneral':
-            collectorFormatFun =  ehubGeneralFormat;
+            formatFun =  ehubGeneralFormat;
             break;
         default:
-            collectorFormatFun =  ehubGeneralFormat;
+            formatFun =  ehubGeneralFormat;
             break;
     }
     
-    async.waterfall([
-        function(callback) {
-            return blobService.getBlobToText(process.env.APP_DL_CONTAINER_NAME, blob.name, callback);
-        },
-        function(blobData, blobReq, blobResp, callback) {
-            try {
-                return ehubCollector(context, JSON.parse(blobData), collectorFormatFun, collectorProcessError, callback);
-            } catch (ex) {
-                return callback(ex);
-            }
-        },
-        function(result, callback) {
-            if (result.skipped === 0) {
-                return blobService.deleteBlob(process.env.APP_DL_CONTAINER_NAME, blob.name, callback);
-            } else {
-                return callback(null, result);
-            }
-        }
-    ], callback);
+    try {
+        ehubCollector(context, JSON.parse(dlblobText), formatFun, processErrorFun, function(error, result) {
```

Comment:
Since we're ignoring result here, maybe we should drop redResult from 
https://github.com/alertlogic/ehub-collector/blob/dlblob/common/ehub_collector.js#L62

And, actually, do we need lines
https://github.com/alertlogic/ehub-collector/blob/dlblob/common/ehub_collector.js#L59-L61
at all there?

Reply:
I'll have a closer look.
But we definitely need these lines https://github.com/alertlogic/ehub-collector/blob/dlblob/common/ehub_collector.js#L59-L61 as this is the actual dead letter blob publishing into the container via function binding.